src/MongoDB/Driver/Manager.cpp

```diff
@@ -192,7 +202,10 @@ static bool hippo_mongo_driver_manager_apply_rp(mongoc_uri_t *uri, const Array o
 	/* Validate that readPreferenceTags are not used with PRIMARY readPreference */
 	if (
 		mongoc_read_prefs_get_mode(new_rp) == MONGOC_READ_PRIMARY &&
-		!bson_empty(mongoc_read_prefs_get_tags(new_rp))
+		(
+			options.exists(s_MongoDBDriverManager_readpreferencetags) ||
+			options.exists(s_MongoDBDriverManager_readPreferenceTags)
+		)
```

Comment:
Why did this need to change? PHPC relies on `!bson_empty(mongoc_read_prefs_get_tags(new_rp)`, which works just fine. By this point, either option should have been assigned if it was present.